erpnext/controllers/selling_controller.py

```diff
@@ -417,6 +418,18 @@ def validate_items(self):
 		from erpnext.controllers.buying_controller import validate_item_type
 		validate_item_type(self, "is_sales_item", "sales")
 
+	def validate_marketing_expense(self):
+		# set discount 100% for marketing order
+		percentage_discount = 0
+		if self.doctype in ("Delivery Note", "Sales Invoice", "Sales Order", "Quotation"):
+			if self.order_type == "Marketing":
+				frappe.db.set_value(self.doctype, self.name, "apply_discount_on", "Grand Total")
+				percentage_discount = 100
+			frappe.db.set_value(self.doctype, self.name, "additional_discount_percentage", percentage_discount)
+			if percentage_discount:
+				frappe.msgprint(_("{0}% discount applied").format(percentage_discount))
```

Comment:
It shouldn't be a dialog, a simple alert should be fine. If we can't do that, then remove this entirely.